### PR TITLE
[AN-5120] adds a Constant/Variable bit ratio toggle (VBR by default) to the Options

### PR DESCRIPTION
**Pull request by makingthematrix**
#### APK
[Download build #8673](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8673/artifact/build/artifact/wire-dev-PR726-8673.apk)
[Download build #8674](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8674/artifact/build/artifact/wire-dev-PR726-8674.apk)
[Download build #8675](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8675/artifact/build/artifact/wire-dev-PR726-8675.apk)
[Download build #8676](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8676/artifact/build/artifact/wire-dev-PR726-8676.apk)

### CODE REVIEW

File: app/src/main/java/com/waz/zclient/pages/main/profile/preferences/OptionsPreferences.java [hunk 1]
```diff
@@ -213,6 +211,13 @@ public Event handlePreferenceChanged(SharedPreferences sharedPreferences, String
         } else if (key.equals(getString(R.string.pref_options_cursor_send_button_key))) {
             boolean sendButtonIsOn = sharedPreferences.getBoolean(key, false);
             event = new ChangedSendButtonSettingEvent(sendButtonIsOn);
+        } else if (key.equals(getString(R.string.pref_options_vbr_key))) {
+            boolean vbrOn = sharedPreferences.getBoolean(key, false);
```
Comment (hakonbo) [inline app/src/main/java/com/waz/zclient/pages/main/profile/preferences/OptionsPreferences.java]: Should use same default value as in XML, so I guess last argument should be changed to `true`

File: app/src/main/java/com/waz/zclient/pages/main/profile/preferences/OptionsPreferences.java [hunk 2]
```diff
@@ -31,13 +31,11 @@
 import com.waz.api.MediaProvider;
 import com.waz.zclient.BaseScalaActivity;
 import com.waz.zclient.R;
+import com.waz.zclient.calling.controllers.CallPermissionsController;
 import com.waz.zclient.controllers.permission.RequestPermissionsObserver;
 import com.waz.zclient.controllers.spotify.SpotifyObserver;
-import com.waz.zclient.core.controllers.tracking.events.settings.ChangedImageDownloadPreferenceEvent;
```
Comment (hakonbo) [inline app/src/main/java/com/waz/zclient/pages/main/profile/preferences/OptionsPreferences.java]: If you copy `codeStyleSettings.xml` in source root into your `.idea` folder and restart the Intellij/Android studio you should get the same code style settings as the rest of us. We tend to not use * imports.

File: app/src/main/java/com/waz/zclient/pages/main/profile/preferences/OptionsPreferences.java [hunk 3]
```diff
@@ -213,6 +211,13 @@ public Event handlePreferenceChanged(SharedPreferences sharedPreferences, String
         } else if (key.equals(getString(R.string.pref_options_cursor_send_button_key))) {
             boolean sendButtonIsOn = sharedPreferences.getBoolean(key, false);
             event = new ChangedSendButtonSettingEvent(sendButtonIsOn);
+        } else if (key.equals(getString(R.string.pref_options_vbr_key))) {
+            boolean vbrOn = sharedPreferences.getBoolean(key, false);
+            CallPermissionsController ctrl = inject(CallPermissionsController.class);
+            if(ctrl != null) {
```
Comment (hakonbo) [inline app/src/main/java/com/waz/zclient/pages/main/profile/preferences/OptionsPreferences.java]: Space after if (will be fixed automatically by the code style xml i believe)

File: app/src/main/scala/com/waz/zclient/calling/controllers/CallPermissionsController.scala [hunk 4]
```diff
@@ -86,6 +86,11 @@ class CallPermissionsController(implicit inj: Injector, cxt: WireContext) extend
       if (withVideo) R.string.calling__cannot_start__no_video_permission__message else R.string.calling__cannot_start__no_permission__message)
   }
 
+  def setVariableBitRateMode(enabled: Boolean): Unit = {
+    val cbrOn = if(enabled) 0 else 1 // in SE it's reversed: we DISABLE cbr instead of enabling vbr and vice versa
+    v3Service.head.map(_.setAudioConstantBitRateEnabled(cbrOn))
```
Comment (hakonbo) [inline app/src/main/scala/com/waz/zclient/calling/controllers/CallPermissionsController.scala]: Is this persisted on the AVS side between restarts?

File: app/src/main/scala/com/waz/zclient/calling/controllers/CurrentCallController.scala [hunk 5]
```diff
@@ -262,6 +264,14 @@ class CurrentCallController(implicit inj: Injector, cxt: WireContext) extends In
     }
   }
 
+  def vbrEnabled: Signal[String] = {
+    val prefService = inject[PreferenceService]
+    prefService.uiPreferenceBooleanSignal(cxt.getResources.getString(R.string.pref_options_vbr_key)).signal.map {
```
Comment (hakonbo) [inline app/src/main/scala/com/waz/zclient/calling/controllers/CurrentCallController.scala]: What will be the default here if user has not set a value?

File: app/src/main/java/com/waz/zclient/pages/main/profile/preferences/OptionsPreferences.java (hunk 2 above)
Comment (makingthematrix) [inline app/src/main/java/com/waz/zclient/pages/main/profile/preferences/OptionsPreferences.java]: Ok, done.

**Comment by Yorgos-V** [comment]
Two things.
When the Variable Bit Rate is ON, there should be no indication on the call UI.
Only when the user decides to switch to Constant Bit Rate.

And a visual design correction.
The setting should be in a subsection that is CALLS inside the options.

**Pull request merged**